src/3d/qgs3dutils.h

```diff
@@ -180,6 +187,62 @@ class _3D_EXPORT Qgs3DUtils
 
     //! Convert from clicked point on the screen to a ray in world coordinates
     static QgsRay3D rayFromScreenPoint( const QPoint &point, const QSize &windowSize, Qt3DRender::QCamera *camera );
+
+    /**
+     * Converts the clicked mouse position to the corresponding 3D world coordinates
+     * \since QGIS 3.24
+     */
+    static QVector3D mouseToWorldPos( double screenX, double screenY, double, const QSize &screenSize, Qt3DRender::QCamera *camera );
+
+    /**
+     * Function used to extract the pitch and yaw (also known as heading) angles from the view vector of the camera [cameraViewCenter - cameraPosition]
+     * \since QGIS 3.24
+     */
+    static void pitchAndYawFromVector( QVector3D vect, double &pitch, double &yaw );
+
+    /**
+     * Converts from view port space to world space
+     * \since QGIS 3.24
+     */
+    static QVector3D worldPosFromDepth( QMatrix4x4 projMatrixInv, QMatrix4x4 viewMatrixInv, float texCoordX, float texCoordY, float depth );
+
+    /**
+     * Converts from screen coordinates to texture coordinates
+     * \see fromTextureToScreenCoordinates()
+     * \since QGIS 3.24
+     */
+    static QVector2D fromScreenToTextureCoordinates( QVector2D screenXY, QSize winSize );
+
+    /**
+     * Converts from texture coordinates coordinates to screen coordinates
+     * \see fromScreenToTextureCoordinates()
+     * \since QGIS 3.24
+     */
+    static QVector2D fromTextureToScreenCoordinates( QVector2D textureXY, QSize winSize );
+
+    /**
+     * Calculates the distance from the point that has a depth value of \a depth and the camera plane as illustrated here:
+     *
+     *  ________o________x             o : Camera view center
+     *  \       |       /|
+     *   \      |      / |
+     *    \     |     /  |             d : the distance returned by this function
+     *     \    |    /   | d
+     *      \   |   /d2  |             x : the point that has depth value of \a depth
+     *       \  |  /     |
+     *        \ | /      |
+     *         \|/       |
+     * ---------c---------
+     *        camera
+     *
+     * \note to calculate the distance from the the point x to the camera center c, you need to divide the returned distance d2
+     * by the dot product between the view vector of the camera and the ray direction coming from the camera center to the desired
+     * point (Make sure both vectors are normalized).
```

Comment:
This docstring confused me a lot - is it for some older version of this function?

If I understand correctly, we convert the depth value from the depth buffer to the actual depth in world coordinates, is that correct? If yes, it would be worth fixing the docstring and maybe also renaming the function - e.g. `depthBufferValueToWorld` or `linearizeDepth` ?

It would be useful to have a reference link (in cpp file) what the formula is about - e.g. https://learnopengl.com/Advanced-OpenGL/Depth-testing

Reply:
I removed thins function since it is used only in Qgs3DUtils::screenPointToWorldPos. Basically after linearizing the depth that value is not the real distance to the camera position but something called the plane perpendicular distnace to the view direction of the camera. The dot product divisiton part is needed to get the correct distance to the camera.
I will try to find some documentation about this, strangly I didn't find many informations about it online...